presto-hdfs-core/src/main/java/com/facebook/presto/hive/HiveFileInfo.java

```diff
@@ -127,7 +130,15 @@ public Map<String, String> getCustomSplitInfo()
 
     public Path getPath()
     {
-        return path;
+        return new Path(path);
+    }
+
+    public long getRetainedSizeInBytes()
+    {
+        long blockLocationsSizeInBytes = blockLocations.stream().map(BlockLocation::getRetainedSizeInBytes).reduce(0L, Long::sum);
+        long extraFileInfoSizeInBytes = extraFileInfo.map(bytes -> bytes.length).orElse(0);
+        long customSplitInfoSizeInBytes = customSplitInfo.entrySet().stream().map(e -> e.getKey().length() + e.getValue().length()).reduce(0, Integer::sum);
+        return INSTANCE_SIZE + path.length() + blockLocationsSizeInBytes + Long.BYTES + Long.BYTES + extraFileInfoSizeInBytes + customSplitInfoSizeInBytes;
```

Comment:
Just curious, why do we need 2 Long.BYTES here? I see that there are 2 long fields, are their sizes not included in INSTANCE_SIZE?

Reply:
good catch. INSTANCE_SIZE already includes the size of primitive variables. fixed it.